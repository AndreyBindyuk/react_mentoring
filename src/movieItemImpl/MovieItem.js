import React from "react";

class MovieItem extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div key={this.props.movie.id} className="movie-item-container">
      <figure class="image"><img alt="#" src={this.props.movie.poster_src}/></figure>
      <div class="movie-title"><span>{this.props.movie.title}</span></div>
      <div className="row">
      <span>{this.props.movie.genre}</span>
      <span style={{color:"#00bbec"}}>{this.props.movie.date}</span>
      </div>
      </div>
    );
  }
}

export default MovieItem;
