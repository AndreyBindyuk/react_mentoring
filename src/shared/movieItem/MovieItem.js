import React from "react";
import { Link } from "react-router-dom";
import "./Movie.css";

export class MovieItem extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {id, title, release_date, genres, poster_path } = this.props.movie;
    return (
      
      <div className="movie-item-container">
        <Link to = {`/movie/${id}`}>
        <figure className="image">
          <img alt="#" src={poster_path} />
        </figure>
        </Link>
        <div className="movie-title">
          <span>{title}</span>
        </div>
        <div className="row">
          <span>{genres["1"]}</span>
          <span style={{ color: "#00bbec" }}>{release_date}</span>
        </div>
      </div>
    );
  }
}

export default MovieItem;
