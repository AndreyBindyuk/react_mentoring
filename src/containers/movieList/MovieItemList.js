import React from "react";
import MovieItem from "../../shared/movieItem/MovieItem";
import MovieStub from "../../MovieStub.json";

class MovieItemList extends React.Component {
  constructor(props) {
    super(props);
    this.state = { movies: MovieStub };
  }

  render() {
    return (
      <div className="movie-list-container">
        {this.state.movies.map(movie => (
          <MovieItem key={movie.id} movie={movie} />
        ))}
      </div>
    );
  }
}

export default MovieItemList;
