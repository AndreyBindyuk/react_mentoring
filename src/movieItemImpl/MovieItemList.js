import React from 'react';
import MovieItem from './MovieItem';
class MovieItemList extends React.Component{
    render(){
        return(
            <div>
            <h1>MovieItemList Component</h1>
            <MovieItem />
            </div>
        );
    }
}

export default MovieItemList;