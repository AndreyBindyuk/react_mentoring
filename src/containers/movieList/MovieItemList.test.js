import React from "react";
import { shallow, mount } from "enzyme";
import MovieItemList from "./MovieItemList";
import MovieItem from "../../shared/movieItem/MovieItem";

describe("List component", () => {
  it("should render components correctly", () => {
    const component = shallow(<MovieItemList />);
    expect(component).toMatchSnapshot();
  });

  it("should render all stub components - Item", () => {
    const component = mount(<MovieItemList />);
    expect(component.find(MovieItem).length).toBe(7);
  });
});
