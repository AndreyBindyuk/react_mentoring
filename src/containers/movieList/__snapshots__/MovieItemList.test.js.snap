// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render components correctly 1`] = `ShallowWrapper {}`;
